dotnet/Controllers/RoutesController.cs

```diff
@@ -395,31 +398,47 @@ public async Task<IActionResult> SendUnsentOrders(string date)
+                                totalOrderCount = totalOrderCount + shipStationOrdersThisOrder.Count;
+                            }
+                            else
+                            {
+                                Console.WriteLine($"{orderId} null?{shipStationOrdersThisOrder == null}");
+                                Console.WriteLine($"{orderId} Count?{shipStationOrdersThisOrder.Count}");
+                            }
+
+                            Console.WriteLine($"ORDER COUNT {orderCount} | {totalOrderCount}");
+                        }
+                        else
+                        {
+                            Console.WriteLine($"vtexOrder = {vtexOrder.OrderId} [{vtexOrder.Status}] ");
                         }
                     }
 
-                    //int batchSize = 100;
-                    //for (int i = 0; i < shipStationOrders.Count(); i += batchSize)
-                    //{
-                    //    var chunk = shipStationOrders.Skip(i).Take(batchSize).ToList();
-                    //    await _vtexAPIService.SendOrdersBatch(chunk);
-                    //}
+                    int batchSize = 100;
+                    for (int i = 0; i < shipStationOrders.Count(); i += batchSize)
+                    {
+                        var chunk = shipStationOrders.Skip(i).Take(batchSize).ToList();
+                        await _vtexAPIService.SendOrdersBatch(chunk);
+                    }
 
-                    _vtexAPIService.SendOrdersBatch(shipStationOrders);
+                    //await _vtexAPIService.SendOrdersBatch(shipStationOrders);
 
                     _context.Vtex.Logger.Warn("SendUnsentOrders", null, sb.ToString());
 
-                    //currentPage = currentPage + 1;
-                    //vtexOrderList = await this._vtexAPIService.ListOrders($"per_page=100&f_status=ready-for-handling&f_creationDate=creationDate:[{dt:yyyy-MM-ddTHH:mm:00Z} TO {dt.AddDays(1):yyyy-MM-ddTHH:mm:00Z}]&page={currentPage}");
-                    vtexOrderList = await this._vtexAPIService.ListOrders($"per_page=100&f_status=ready-for-handling&f_creationDate=creationDate:[{dt:yyyy-MM-ddTHH:mm:00Z} TO {dt.AddDays(1):yyyy-MM-ddTHH:mm:00Z}]");
+                    currentPage = currentPage + 1;
+                    vtexOrderList = await this._vtexAPIService.ListOrders($"per_page=100&f_status=ready-for-handling&f_creationDate=creationDate:[{dt:yyyy-MM-ddTHH:mm:00Z} TO {dt.AddDays(1):yyyy-MM-ddTHH:mm:00Z}]&page={currentPage}");
+                    //vtexOrderList = await this._vtexAPIService.ListOrders($"per_page=100&f_status=ready-for-handling&f_creationDate=creationDate:[{dt:yyyy-MM-ddTHH:mm:00Z} TO {dt.AddDays(1):yyyy-MM-ddTHH:mm:00Z}]");
                 }
             }
-            while (vtexOrderList != null && vtexOrderList.List != null && vtexOrderList.List.Length > 0);
-            //while (currentPage <= totalPages);
+            while (currentPage <= totalPages);
```

Comment:
```suggestion
            while (currentPage < totalPages);
```
`currentPage` should never be equal to `totalPages`, right? Since `currentPage` starts at 0

Reply:
zero is the default value, if the list request is not null it will be set to the current page of the response:
currentPage = vtexOrderList.Paging.CurrentPage;

Reply:
Ok gotcha